src/scripts/main.js

```diff
@@ -1,3 +1,218 @@
+  </label>`;
+const button = '<button>Save to table</button>';
+
+form.insertAdjacentHTML('afterbegin', button);
+form.insertAdjacentHTML('afterbegin', formSalary);
+form.insertAdjacentHTML('afterbegin', formAge);
+form.insertAdjacentHTML('afterbegin', formCountry);
+form.insertAdjacentHTML('afterbegin', formPosition);
+form.insertAdjacentHTML('afterbegin', formName);
+
+const inputs = form.querySelectorAll('input');
+
+inputs.forEach((input) => input.setAttribute('required', ''));
+
+const createNotification = (text, className) => {
+  const notification = document.createElement('div');
+
+  notification.classList.add('notification', className);
+  notification.setAttribute('data-qa', 'notification');
+
+  const desctiption = document.createElement('h1');
+
+  desctiption.classList.add('title');
+  desctiption.textContent = text;
+  notification.appendChild(desctiption);
+
+  return notification;
+};
+
+form.addEventListener('submit', (e) => {
+  e.preventDefault();
+
+  // eslint-disable-next-line no-shadow
+  const name = form.querySelector('[data-qa="name"]').value;
+  const position = form.querySelector('[data-qa="position"]').value;
+  const age = form.querySelector('[data-qa="age"]').value;
+  const salary = form.querySelector('[data-qa="salary"]').value;
+  const office = form.querySelector('[data-qa="office"]').value;
+
+  let isValid = true;
+  let message = '';
+
+  if (name.trim().length < 4) {
+    message = 'Name value has less than 4 letters';
+    isValid = false;
+  } else if (position.trim().length === 0) {
+    message = 'Position value is invalid';
+    isValid = false;
+  } else if (age < 18 || age > 90) {
+    message = 'Age value is less than 18 or more than 90';
+    isValid = false;
+  }
+
+  if (!isValid) {
+    document.body.append(createNotification(message, 'error'));
+  } else {
+    document.body.append(
+      createNotification('Employee successfully added', 'success'),
+    );
+
+    const newPerson = [
+      name.trim(),
+      position.trim(),
+      office,
+      age,
+      `$${(+(salary / 1000)).toFixed(3).replace('.', ',')}`,
+    ];
+
+    const newRow = tbody.insertRow(-1);
+
+    newPerson.forEach((item, index) => {
+      const cell = newRow.insertCell(index);
+
+      cell.innerText = item;
+    });
+
+    form.reset();
+  }
+
+  setTimeout(() => {
+    const notifications = document.querySelectorAll('.notification');
+
+    notifications.forEach((notification) => {
+      notification.parentNode.removeChild(notification);
+    });
+  }, 3000);
+});
+
+// eslint-disable-next-line no-shadow
+tbody.addEventListener('dblclick', (e) => {
+  const cell = e.target;
+  let editedCell = cell.textContent;
+
+  cell.innerHTML = `<input class='cell-input' value='${editedCell}' />`;
+
+  const input = e.target.querySelector('.cell-input');
+
+  input.focus();
+
+  // eslint-disable-next-line no-shadow
+  const saveChanges = (e) => {
+    if (e.type === 'blur' || (e.type === 'keyup' && e.key === 'Enter')) {
```

Comment:
Would be nice to properly format salary after editing (you can move it to some function and reuse when editing and adding new row)

![image](https://github.com/mate-academy/js_employees_table_DOM/assets/86923089/8dad89b3-292f-409f-8004-636f29eb544e)